### ISSUE TITLE
Why envoy behaves differently when envoy adds an error tag to trace according to HTTP and GRPC status?

### ISSUE THREAD

**Issue by fukuda151** [envoyproxy/envoy#15883]
*Description*:
Envoy adds an error tag to trace when HTTP status is 5xx (NOT when HTTP status is 4xx). On the other hand, envoy adds an error tag to trace when GRPC status is not "OK". 

I think envoy should not adds an error tag to trace when GRPC status is such as "NOT_FOUND" or "PERMISSION_DENIED" because they are equivalent to HTTP status 4xx.

Is there any intention for this difference?

*Relevant Links*:
https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/observability/tracing#what-data-each-trace-contains

**Comment by snowp** [comment]
I don't think this is intentional, we'd probably be up for changing to be consistent here.

**Comment by jakubednarek** [comment]
I had a same issue and created a PR with a fix: https://github.com/envoyproxy/envoy/pull/16078. I would appreciate a review if this is an acceptable change.
